python/lancedb/rerankers/linear_combination.py

```diff
@@ -0,0 +1,79 @@
+from .base import Reranker
+import pyarrow as pa
+
+
+class LinearCombinationReranker(Reranker):
+    """
+    Reranks the results using a linear combination of the scores from the vector and FTS search.
+    For missing scores, fill with `fill` value.
+    Parameters
+    ----------
+    weight : float, default 0.5
+        The weight to give to the vector score. Must be between 0 and 1.
+    fill : float, default 1.0
+        The score to give to results that are only in one of the two result sets.
+    """
+
+    def __init__(self, weight: float = 0.5, fill: float = 1.0):
+        if weight < 0 or weight > 1:
+            raise ValueError("weight must be between 0 and 1.")
+        self.weight = weight
+        self.fill = fill
+
+    def rerank_hybrid(
+        self,
+        query_builder: "lancedb.HybridQueryBuilder",  # noqa: F821
+        vector_results: pa.Table,
+        fts_results: pa.Table,
+    ):
+        combined_results = self.merge_results(vector_results, fts_results, self.fill)
+
+        return combined_results
+
+    def merge_results(
+        self, vector_results: pa.Table, fts_results: pa.Table, fill: float
+    ):
+        # If both are empty then just return an empty table
+        if len(vector_results) == 0 and len(fts_results) == 0:
+            return vector_results
+        # If one is empty then return the other
+        if len(vector_results) == 0:
+            return fts_results
+        if len(fts_results) == 0:
+            return vector_results
+
+        # sort both input tables on _rowid
+        combined_list = []
+        vector_list = vector_results.sort_by("_rowid").to_pylist()
+        fts_list = fts_results.sort_by("_rowid").to_pylist()
+        i, j = 0, 0
+        while i < len(vector_list):
```

Comment:
this is going to be monster slow unfortunately have you timed this? Didn't I already have an implementation for this using pyarrow / numpy in my PR?

Reply:
@changhiskhan uhh this is from your PR only. Can you check if you had implemented this in pyarrow locally and maybe not pushed?